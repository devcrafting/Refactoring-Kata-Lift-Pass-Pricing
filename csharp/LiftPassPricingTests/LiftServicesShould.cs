using System;
using MySql.Data.MySqlClient;
using Xunit;
using FsCheck.Xunit;
using FluentAssertions;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;

public class LiftServicesShould
{
    private readonly IStoreLiftPricer liftPricerRepository;

    public Random random { get; }

    public LiftServicesShould()
    {
        var connection = new MySqlConnection
            {
                ConnectionString = @"Database=lift_pass;Data Source=localhost;User Id=root;Password=mysql"
            };
        connection.Open();
        liftPricerRepository = new LiftPricerRepository(connection);
        random = new Random(1); 
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void ReturnGoldenMaster()
    {
        var types = new [] { "night", "1jour" };

        var result =
            Enumerable.Range(0, 99)
                .Cast<int?>()
                .Concat(new int?[] { null })
                .SelectMany(age => types.SelectMany(type => {
                    var date = new DateTime(random.Next(2017, 2019), random.Next(1, 12), random.Next(1, 28));
                    return new [] {
                        $"- {age}, {type}, {date.ToString("yyyy-MM-dd")}\n",
                        liftPricerRepository.Get(type, date).GetPrice(age) + "\n",
                        liftPricerRepository.Get(type, DateTime.Parse("2019-02-18")).GetPrice(age) + "\n"
                    };
                }));

        Approvals.Verify(string.Concat(result));
    }
}