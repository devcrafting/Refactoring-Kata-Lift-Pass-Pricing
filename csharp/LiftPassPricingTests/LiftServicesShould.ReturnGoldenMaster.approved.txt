- 0, night, 2017-02-13
{ "cost": 0}
{ "cost": 0}
- 0, 1jour, 2018-08-12
{ "cost": 0}
{ "cost": 0}
- 1, night, 2017-11-03
{ "cost": 0}
{ "cost": 0}
- 1, 1jour, 2018-01-07
{ "cost": 0}
{ "cost": 0}
- 2, night, 2017-11-19
{ "cost": 0}
{ "cost": 0}
- 2, 1jour, 2018-04-17
{ "cost": 0}
{ "cost": 0}
- 3, night, 2018-08-26
{ "cost": 0}
{ "cost": 0}
- 3, 1jour, 2017-02-11
{ "cost": 0}
{ "cost": 0}
- 4, night, 2018-02-22
{ "cost": 0}
{ "cost": 0}
- 4, 1jour, 2017-10-24
{ "cost": 0}
{ "cost": 0}
- 5, night, 2018-08-19
{ "cost": 0}
{ "cost": 0}
- 5, 1jour, 2017-11-22
{ "cost": 0}
{ "cost": 0}
- 6, night, 2018-01-14
{ "cost": 19}
{ "cost": 19}
- 6, 1jour, 2018-04-27
{ "cost": 25}
{ "cost": 25}
- 7, night, 2018-05-12
{ "cost": 19}
{ "cost": 19}
- 7, 1jour, 2018-08-13
{ "cost": 25}
{ "cost": 25}
- 8, night, 2017-04-15
{ "cost": 19}
{ "cost": 19}
- 8, 1jour, 2017-04-19
{ "cost": 25}
{ "cost": 25}
- 9, night, 2017-02-26
{ "cost": 19}
{ "cost": 19}
- 9, 1jour, 2017-11-14
{ "cost": 25}
{ "cost": 25}
- 10, night, 2018-01-04
{ "cost": 19}
{ "cost": 19}
- 10, 1jour, 2017-01-09
{ "cost": 25}
{ "cost": 25}
- 11, night, 2018-04-27
{ "cost": 19}
{ "cost": 19}
- 11, 1jour, 2018-09-06
{ "cost": 25}
{ "cost": 25}
- 12, night, 2017-02-12
{ "cost": 19}
{ "cost": 19}
- 12, 1jour, 2018-05-21
{ "cost": 25}
{ "cost": 25}
- 13, night, 2018-02-14
{ "cost": 19}
{ "cost": 19}
- 13, 1jour, 2017-01-13
{ "cost": 25}
{ "cost": 25}
- 14, night, 2017-05-10
{ "cost": 19}
{ "cost": 19}
- 14, 1jour, 2017-07-23
{ "cost": 25}
{ "cost": 25}
- 15, night, 2018-09-03
{ "cost": 19}
{ "cost": 19}
- 15, 1jour, 2017-10-07
{ "cost": 35}
{ "cost": 35}
- 16, night, 2018-06-09
{ "cost": 19}
{ "cost": 19}
- 16, 1jour, 2017-02-11
{ "cost": 35}
{ "cost": 35}
- 17, night, 2017-05-16
{ "cost": 19}
{ "cost": 19}
- 17, 1jour, 2017-01-05
{ "cost": 35}
{ "cost": 35}
- 18, night, 2017-05-21
{ "cost": 19}
{ "cost": 19}
- 18, 1jour, 2017-02-23
{ "cost": 35}
{ "cost": 35}
- 19, night, 2018-04-27
{ "cost": 19}
{ "cost": 19}
- 19, 1jour, 2018-10-19
{ "cost": 35}
{ "cost": 35}
- 20, night, 2018-06-19
{ "cost": 19}
{ "cost": 19}
- 20, 1jour, 2017-01-01
{ "cost": 35}
{ "cost": 35}
- 21, night, 2017-01-08
{ "cost": 19}
{ "cost": 19}
- 21, 1jour, 2017-01-26
{ "cost": 35}
{ "cost": 35}
- 22, night, 2017-10-27
{ "cost": 19}
{ "cost": 19}
- 22, 1jour, 2017-10-19
{ "cost": 35}
{ "cost": 35}
- 23, night, 2018-01-09
{ "cost": 19}
{ "cost": 19}
- 23, 1jour, 2017-09-05
{ "cost": 35}
{ "cost": 35}
- 24, night, 2017-08-18
{ "cost": 19}
{ "cost": 19}
- 24, 1jour, 2017-01-16
{ "cost": 23}
{ "cost": 35}
- 25, night, 2017-03-17
{ "cost": 19}
{ "cost": 19}
- 25, 1jour, 2018-02-16
{ "cost": 35}
{ "cost": 35}
- 26, night, 2018-04-10
{ "cost": 19}
{ "cost": 19}
- 26, 1jour, 2018-02-27
{ "cost": 35}
{ "cost": 35}
- 27, night, 2018-01-10
{ "cost": 19}
{ "cost": 19}
- 27, 1jour, 2018-08-09
{ "cost": 35}
{ "cost": 35}
- 28, night, 2018-10-13
{ "cost": 19}
{ "cost": 19}
- 28, 1jour, 2017-09-22
{ "cost": 35}
{ "cost": 35}
- 29, night, 2017-09-22
{ "cost": 19}
{ "cost": 19}
- 29, 1jour, 2018-01-10
{ "cost": 35}
{ "cost": 35}
- 30, night, 2017-10-02
{ "cost": 19}
{ "cost": 19}
- 30, 1jour, 2018-11-24
{ "cost": 35}
{ "cost": 35}
- 31, night, 2017-10-18
{ "cost": 19}
{ "cost": 19}
- 31, 1jour, 2017-05-15
{ "cost": 23}
{ "cost": 35}
- 32, night, 2017-05-24
{ "cost": 19}
{ "cost": 19}
- 32, 1jour, 2017-04-19
{ "cost": 35}
{ "cost": 35}
- 33, night, 2018-03-27
{ "cost": 19}
{ "cost": 19}
- 33, 1jour, 2017-05-05
{ "cost": 35}
{ "cost": 35}
- 34, night, 2017-11-14
{ "cost": 19}
{ "cost": 19}
- 34, 1jour, 2018-03-12
{ "cost": 23}
{ "cost": 35}
- 35, night, 2018-02-07
{ "cost": 19}
{ "cost": 19}
- 35, 1jour, 2017-02-08
{ "cost": 35}
{ "cost": 35}
- 36, night, 2018-02-03
{ "cost": 19}
{ "cost": 19}
- 36, 1jour, 2017-06-23
{ "cost": 35}
{ "cost": 35}
- 37, night, 2018-05-12
{ "cost": 19}
{ "cost": 19}
- 37, 1jour, 2018-11-10
{ "cost": 35}
{ "cost": 35}
- 38, night, 2018-01-12
{ "cost": 19}
{ "cost": 19}
- 38, 1jour, 2017-08-23
{ "cost": 35}
{ "cost": 35}
- 39, night, 2017-01-14
{ "cost": 19}
{ "cost": 19}
- 39, 1jour, 2018-07-01
{ "cost": 35}
{ "cost": 35}
- 40, night, 2017-06-18
{ "cost": 19}
{ "cost": 19}
- 40, 1jour, 2017-07-08
{ "cost": 35}
{ "cost": 35}
- 41, night, 2017-04-07
{ "cost": 19}
{ "cost": 19}
- 41, 1jour, 2018-05-05
{ "cost": 35}
{ "cost": 35}
- 42, night, 2018-04-20
{ "cost": 19}
{ "cost": 19}
- 42, 1jour, 2017-08-26
{ "cost": 35}
{ "cost": 35}
- 43, night, 2018-02-01
{ "cost": 19}
{ "cost": 19}
- 43, 1jour, 2017-10-05
{ "cost": 35}
{ "cost": 35}
- 44, night, 2017-08-14
{ "cost": 19}
{ "cost": 19}
- 44, 1jour, 2017-08-03
{ "cost": 35}
{ "cost": 35}
- 45, night, 2018-01-15
{ "cost": 19}
{ "cost": 19}
- 45, 1jour, 2017-02-02
{ "cost": 35}
{ "cost": 35}
- 46, night, 2017-06-23
{ "cost": 19}
{ "cost": 19}
- 46, 1jour, 2017-08-17
{ "cost": 35}
{ "cost": 35}
- 47, night, 2017-04-19
{ "cost": 19}
{ "cost": 19}
- 47, 1jour, 2017-06-10
{ "cost": 35}
{ "cost": 35}
- 48, night, 2017-03-11
{ "cost": 19}
{ "cost": 19}
- 48, 1jour, 2018-02-13
{ "cost": 35}
{ "cost": 35}
- 49, night, 2017-03-16
{ "cost": 19}
{ "cost": 19}
- 49, 1jour, 2017-02-26
{ "cost": 35}
{ "cost": 35}
- 50, night, 2018-10-19
{ "cost": 19}
{ "cost": 19}
- 50, 1jour, 2017-10-09
{ "cost": 23}
{ "cost": 35}
- 51, night, 2018-06-06
{ "cost": 19}
{ "cost": 19}
- 51, 1jour, 2018-11-06
{ "cost": 35}
{ "cost": 35}
- 52, night, 2017-07-14
{ "cost": 19}
{ "cost": 19}
- 52, 1jour, 2017-11-16
{ "cost": 35}
{ "cost": 35}
- 53, night, 2017-09-05
{ "cost": 19}
{ "cost": 19}
- 53, 1jour, 2017-04-12
{ "cost": 35}
{ "cost": 35}
- 54, night, 2018-03-25
{ "cost": 19}
{ "cost": 19}
- 54, 1jour, 2017-01-24
{ "cost": 35}
{ "cost": 35}
- 55, night, 2017-08-08
{ "cost": 19}
{ "cost": 19}
- 55, 1jour, 2018-05-21
{ "cost": 23}
{ "cost": 35}
- 56, night, 2018-11-14
{ "cost": 19}
{ "cost": 19}
- 56, 1jour, 2017-07-01
{ "cost": 35}
{ "cost": 35}
- 57, night, 2017-05-07
{ "cost": 19}
{ "cost": 19}
- 57, 1jour, 2017-06-17
{ "cost": 35}
{ "cost": 35}
- 58, night, 2018-07-08
{ "cost": 19}
{ "cost": 19}
- 58, 1jour, 2017-01-11
{ "cost": 35}
{ "cost": 35}
- 59, night, 2018-08-15
{ "cost": 19}
{ "cost": 19}
- 59, 1jour, 2017-06-18
{ "cost": 35}
{ "cost": 35}
- 60, night, 2017-07-12
{ "cost": 19}
{ "cost": 19}
- 60, 1jour, 2017-03-08
{ "cost": 35}
{ "cost": 35}
- 61, night, 2018-03-05
{ "cost": 19}
{ "cost": 19}
- 61, 1jour, 2018-06-27
{ "cost": 35}
{ "cost": 35}
- 62, night, 2017-02-06
{ "cost": 19}
{ "cost": 19}
- 62, 1jour, 2017-11-26
{ "cost": 35}
{ "cost": 35}
- 63, night, 2017-06-20
{ "cost": 19}
{ "cost": 19}
- 63, 1jour, 2017-01-13
{ "cost": 35}
{ "cost": 35}
- 64, night, 2018-05-19
{ "cost": 19}
{ "cost": 19}
- 64, 1jour, 2018-02-21
{ "cost": 35}
{ "cost": 35}
- 65, night, 2017-08-15
{ "cost": 8}
{ "cost": 8}
- 65, 1jour, 2018-11-27
{ "cost": 27}
{ "cost": 27}
- 66, night, 2018-09-05
{ "cost": 8}
{ "cost": 8}
- 66, 1jour, 2018-04-07
{ "cost": 27}
{ "cost": 27}
- 67, night, 2017-03-04
{ "cost": 8}
{ "cost": 8}
- 67, 1jour, 2017-11-25
{ "cost": 27}
{ "cost": 27}
- 68, night, 2017-08-18
{ "cost": 8}
{ "cost": 8}
- 68, 1jour, 2018-01-27
{ "cost": 27}
{ "cost": 27}
- 69, night, 2018-02-14
{ "cost": 8}
{ "cost": 8}
- 69, 1jour, 2018-09-22
{ "cost": 27}
{ "cost": 27}
- 70, night, 2018-11-04
{ "cost": 8}
{ "cost": 8}
- 70, 1jour, 2017-06-21
{ "cost": 27}
{ "cost": 27}
- 71, night, 2017-03-05
{ "cost": 8}
{ "cost": 8}
- 71, 1jour, 2017-06-06
{ "cost": 27}
{ "cost": 27}
- 72, night, 2018-03-05
{ "cost": 8}
{ "cost": 8}
- 72, 1jour, 2018-09-24
{ "cost": 18}
{ "cost": 27}
- 73, night, 2017-04-15
{ "cost": 8}
{ "cost": 8}
- 73, 1jour, 2018-09-12
{ "cost": 27}
{ "cost": 27}
- 74, night, 2017-07-18
{ "cost": 8}
{ "cost": 8}
- 74, 1jour, 2018-11-22
{ "cost": 27}
{ "cost": 27}
- 75, night, 2017-08-01
{ "cost": 8}
{ "cost": 8}
- 75, 1jour, 2017-04-09
{ "cost": 27}
{ "cost": 27}
- 76, night, 2017-01-19
{ "cost": 8}
{ "cost": 8}
- 76, 1jour, 2017-09-19
{ "cost": 27}
{ "cost": 27}
- 77, night, 2018-11-05
{ "cost": 8}
{ "cost": 8}
- 77, 1jour, 2017-10-21
{ "cost": 27}
{ "cost": 27}
- 78, night, 2018-05-10
{ "cost": 8}
{ "cost": 8}
- 78, 1jour, 2018-07-13
{ "cost": 27}
{ "cost": 27}
- 79, night, 2017-09-07
{ "cost": 8}
{ "cost": 8}
- 79, 1jour, 2018-04-26
{ "cost": 27}
{ "cost": 27}
- 80, night, 2017-11-07
{ "cost": 8}
{ "cost": 8}
- 80, 1jour, 2017-09-23
{ "cost": 27}
{ "cost": 27}
- 81, night, 2017-05-25
{ "cost": 8}
{ "cost": 8}
- 81, 1jour, 2018-05-20
{ "cost": 27}
{ "cost": 27}
- 82, night, 2017-11-14
{ "cost": 8}
{ "cost": 8}
- 82, 1jour, 2018-01-23
{ "cost": 27}
{ "cost": 27}
- 83, night, 2017-07-21
{ "cost": 8}
{ "cost": 8}
- 83, 1jour, 2018-08-16
{ "cost": 27}
{ "cost": 27}
- 84, night, 2017-09-03
{ "cost": 8}
{ "cost": 8}
- 84, 1jour, 2018-02-15
{ "cost": 27}
{ "cost": 27}
- 85, night, 2017-05-21
{ "cost": 8}
{ "cost": 8}
- 85, 1jour, 2018-03-24
{ "cost": 27}
{ "cost": 27}
- 86, night, 2017-04-04
{ "cost": 8}
{ "cost": 8}
- 86, 1jour, 2017-04-12
{ "cost": 27}
{ "cost": 27}
- 87, night, 2018-01-27
{ "cost": 8}
{ "cost": 8}
- 87, 1jour, 2018-05-18
{ "cost": 27}
{ "cost": 27}
- 88, night, 2018-05-20
{ "cost": 8}
{ "cost": 8}
- 88, 1jour, 2017-05-19
{ "cost": 27}
{ "cost": 27}
- 89, night, 2017-03-07
{ "cost": 8}
{ "cost": 8}
- 89, 1jour, 2018-11-27
{ "cost": 27}
{ "cost": 27}
- 90, night, 2018-07-08
{ "cost": 8}
{ "cost": 8}
- 90, 1jour, 2017-09-02
{ "cost": 27}
{ "cost": 27}
- 91, night, 2018-06-20
{ "cost": 8}
{ "cost": 8}
- 91, 1jour, 2018-05-20
{ "cost": 27}
{ "cost": 27}
- 92, night, 2018-11-10
{ "cost": 8}
{ "cost": 8}
- 92, 1jour, 2017-01-19
{ "cost": 27}
{ "cost": 27}
- 93, night, 2018-09-11
{ "cost": 8}
{ "cost": 8}
- 93, 1jour, 2017-07-20
{ "cost": 27}
{ "cost": 27}
- 94, night, 2018-08-26
{ "cost": 8}
{ "cost": 8}
- 94, 1jour, 2017-06-01
{ "cost": 27}
{ "cost": 27}
- 95, night, 2018-09-11
{ "cost": 8}
{ "cost": 8}
- 95, 1jour, 2017-07-19
{ "cost": 27}
{ "cost": 27}
- 96, night, 2017-06-08
{ "cost": 8}
{ "cost": 8}
- 96, 1jour, 2018-08-26
{ "cost": 27}
{ "cost": 27}
- 97, night, 2018-04-01
{ "cost": 8}
{ "cost": 8}
- 97, 1jour, 2017-08-13
{ "cost": 27}
{ "cost": 27}
- 98, night, 2017-10-07
{ "cost": 8}
{ "cost": 8}
- 98, 1jour, 2018-01-04
{ "cost": 27}
{ "cost": 27}
- , night, 2018-04-27
{ "cost": 0}
{ "cost": 0}
- , 1jour, 2018-07-01
{ "cost": 35}
{ "cost": 35}
